carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/devices/WebBrowser.kt

```diff
@@ -0,0 +1,68 @@
+@file:Suppress("NON_EXPORTABLE_TYPE")
+
+package dk.cachet.carp.common.application.devices
+
+import dk.cachet.carp.common.application.Trilean
+import dk.cachet.carp.common.application.URL
+import dk.cachet.carp.common.application.data.DataType
+import dk.cachet.carp.common.application.sampling.DataTypeSamplingSchemeMap
+import dk.cachet.carp.common.application.sampling.SamplingConfiguration
+import dk.cachet.carp.common.application.tasks.TaskConfigurationList
+import dk.cachet.carp.common.application.toTrilean
+import dk.cachet.carp.common.infrastructure.serialization.NotSerializable
+import kotlinx.serialization.Required
+import kotlinx.serialization.Serializable
+import kotlin.js.JsExport
+import kotlin.reflect.KClass
+
+
+@Serializable
+@JsExport
+data class WebBrowser(
+        val possibleTypes: Set<String>,
```

Comment:
I had no clue what this was until I read the full code.

Documentation needed, and a rename. I suggest `supportedBrowsers`. But, there is also a design issue. If you have no reason to believe a given browser _wouldn't_ be supported (which should really be the predominant scenario if good web design is assumed), you'd have to provide an exhaustive list here. So in the _majority_ of use cases, you need to provide/know a full list, and even then, a study would fail early if new browsers arise, or they change name. Thus, instead you probably want `unsupportedBrowsers` and a default empty list.

In addition, this (and in `WebBrowserDeviceRegistration`) needs clear instructions on what to pass here. I think I previously suggested this likely relates to HTTP header info (and p.s., if that is the case, `URL` is probably too broad of an abstraction, and you want to limit this to HTTP).

That is reason enough to add a `Browser` type which encapsulates this knowledge for existing browsers (and allows parsing it from the information in the string), so API users don't need to look up what this is for each browser, but which still provides the option to pass `Other`, for forwards compatability. There is instant reuse across `possibleTypes`, and `browser` in the registration, so this isn't premature. And, in front-end representations (this is multiplatform, so also relevant for JS code), it will help showing e.g. pie charts.

Reply:
`WebBrowser`. And given the name ambiguity, it also warrants rethinking the name for the device configuration. Sounds like `Website` may be more appropriate. The client runs on a website, which you access with the browser (device registration information).